lib/libp2p_port.ex

```diff
@@ -187,25 +187,16 @@ defmodule LambdaEthereumConsensus.Libp2pPort do
   Subscribes to the given topic. After this, messages published to the topic
   will be received by `self()`.
   """
-  @spec subscribe_to_topic(GenServer.server(), String.t()) :: :ok | {:error, String.t()}
-  def subscribe_to_topic(pid \\ __MODULE__, topic_name) do
+  @spec subscribe_to_topic(GenServer.server(), String.t(), atom()) :: :ok | {:error, String.t()}
+  def subscribe_to_topic(pid \\ __MODULE__, topic_name, module) do
     :telemetry.execute([:port, :message], %{}, %{
       function: "subscribe_to_topic",
       direction: "elixir->"
     })
 
-    call_command(pid, {:subscribe, %SubscribeToTopic{name: topic_name}})
-  end
+    GenServer.cast(__MODULE__, {:new_subscriptor, String.to_atom(topic_name), module})
```

Comment:
This should send the message to `pid` instead of `__MODULE__`. With that we can re-add the removed tests.